### PR TITLE
Feat/validator

### PR DESCRIPTION
**Pull request by codeflyer**

### CODE REVIEW

File: pickup/lib/pickupBatch.js [hunk 1]
```diff
@@ -80,10 +94,17 @@ export async function pickupBatch (messages, { ipfsApiUrl, createS3Uploader, s3,
       if (downloadError.code === ERROR_TIMEOUT ||
         currentRetry >= maxRetry
       ) {
+        const errorMessage = currentRetry >= maxRetry ? 'Max retry' : 'Download timeout'
```
Comment (simone-sanfratello) [inline pickup/lib/pickupBatch.js:97]: I'd add "Download timeout, retrying"

File: stacks/PickupStack.ts [hunk 2]
```diff
@@ -117,8 +141,33 @@ export function PickupStack ({ app, stack }: StackContext): void {
     basicApi.bucket.cdk.bucket.grantReadWrite(service.taskDefinition.taskRole)
     basicApi.dynamoDbTable.cdk.table.grantReadWriteData(service.taskDefinition.taskRole)
     basicApi.queue.cdk.queue.grantConsumeMessages(service.taskDefinition.taskRole)
+
+    if (process.env.USE_VALIDATION === 'VALIDATE') {
```
Comment (simone-sanfratello) [inline stacks/PickupStack.ts:145]: Which values `process.env.USE_VALIDATION` can be? Looks like a boolean

File: validator/lib/validateCars.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    carInfo = await getCarInfo({ cid, carStream: s3Object.Body })
+
+    if (carInfo.errors.length) {
+      errors = carInfo.errors
+      throw new Error('Car not valid')
+    }
+
+    logger.info({ cid, key }, 'Car valid')
+
+    await updatePinStatus({ dynamo, table: dynamoTable, cid, size, status: 'pinned' })
```
Comment (simone-sanfratello) [inline validator/lib/validateCars.js:52]: I'd avoid writing ops on the validator, the function says `validateCar` but it also writes, looks like is doing too much

File: stacks/PickupStack.ts (hunk 2 above)
Comment (codeflyer) [reply stacks/PickupStack.ts:145]: Is a boolean, but I don't manage the ENV, I need to be sure that's enabled only on the `VALIDATE` value

File: pickup/lib/pickupBatch.js (hunk 1 above)
Comment (codeflyer) [reply pickup/lib/pickupBatch.js:97]: doesn't retry in case of timeout

File: validator/lib/validateCars.js (hunk 3 above)
Comment (codeflyer) [reply validator/lib/validateCars.js]: updated the logic and separated the responsibilities

**Review by simone-sanfratello — APPROVED** [review]

**Comment by olizilla** [comment]
@codeflyer can you add details to the description of this PR (and all PRs on w3 projects) so we can learn about the motivations for a change.

What happens in pickup today without this validator work?

**Comment by olizilla** [comment]
Not all CARS contain unix-fs encoded data. When a user pins a CID it could be any codec or hash that IPLD knows about.

**Pull request merged**